File: src/leetCode/easy/Solution14.java
package leetCode.easy;

class Solution14 {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        String result = strs[0];
        for (int i = 1; i < strs.length; i++)
            result = getCommonPrefix(result, strs[i]);
        return result;
    }

    private String getCommonPrefix(String word1, String word2) {
        StringBuffer result = new StringBuffer();
        for (int i = 0; i < Math.min(word1.length(), word2.length()); i++)
            if (word1.charAt(i) == word2.charAt(i))
                result.append(word1.charAt(i));
            else
                return result.toString();
        return result.toString();
    }
}

File: src/leetcode/easy/Solution14.cs
// https://leetcode.com/problems/longest-common-prefix/ 
public class Solution14 {
    public string LongestCommonPrefix(String[] strs) {
        
        if(strs == null || strs.Length == 0)
        {
            return "";
        }

        StringBuilder str = new StringBuilder();
        Array.Sort(strs);
        var first = strs[0];
        var last = strs[strs.Length - 1];

        for(int i=0; i<Math.Min(first.Length, last.Length); i++)
        {
            if(first[i] != last[i])
            {
                return str.ToString();
            }
            str.Append(first[i]);
        }
        return str.ToString();
    }
}
